web/src/condense.ts

```diff
@@ -18,38 +19,46 @@ This library implements two related, similar concepts:
 
 */
 
-function show_more_link($row) {
+function show_more_link($row: JQuery): void {
     $row.find(".message_condenser").hide();
     $row.find(".message_expander").show();
 }
 
-function show_condense_link($row) {
+function show_condense_link($row: JQuery): void {
     $row.find(".message_expander").hide();
     $row.find(".message_condenser").show();
 }
 
-function condense_row($row) {
+function condense_row($row: JQuery): void {
     const $content = $row.find(".message_content");
     $content.addClass("condensed");
     show_more_link($row);
 }
 
-function uncondense_row($row) {
+function uncondense_row($row: JQuery): void {
     const $content = $row.find(".message_content");
     $content.removeClass("condensed");
     show_condense_link($row);
 }
 
-export function uncollapse($row) {
+export function uncollapse($row: JQuery) {
     assert(message_lists.current !== undefined);
+    assert(message_lists.home !== undefined);
+    const id = rows.id($row)
+    if (id === undefined) {
+        return;
+    }
 
     // Uncollapse a message, restoring the condensed message "Show more" or
     // "Show less" button if necessary.
-    const message = message_lists.current.get(rows.id($row));
+    const message = message_lists.current.get(id);
+    if (message === undefined) {
```

Comment:
There is no need to check if the message is undefined or not here is why.
1) we are checking if we are doing assertation above (line 45) to see if current is undefined or not.
which means, if message_lists exists then there exists a valid id

Reply:
The `get ` function returns `Message | undefined`.

Reply:
` message_flags.save_uncollapsed(message);`

@LaPulgaaa I added the extra type check because, on the line of code, you get a type error like this when you hover on message.

> Argument of type '(Omit<Omit<{ avatar_url: string | null; client: string; content: string; content_type: "text/html"; display_recipient: import("/home/danajayi/zulip/web/src/message_store").DisplayRecipient; ... 13 more ...; flags: string[]; } & { ...; } & import("/home/danajayi/zulip/web/src/message_store").MatchedMessage & { ...; }...' is not assignable to parameter of type '(Omit<Omit<{ avatar_url: string | null; client: string; content: string; content_type: "text/html"; display_recipient: import("/home/danajayi/zulip/web/src/message_store").DisplayRecipient; ... 13 more ...; flags: string[]; } & { ...; } & import("/home/danajayi/zulip/web/src/message_store").MatchedMessage & { ...; }...'.
>   Type 'undefined' is not assignable to type '(Omit<Omit<{ avatar_url: string | null; client: string; content: string; content_type: "text/html"; display_recipient: DisplayRecipient; edit_history?: MessageEditHistoryEntry[] | undefined; ... 12 more ...; flags: string[]; } & { ...; } & MatchedMessage & { ...; }, "flags"> & { ...; } & { ...; }, "reactions"> & { ....'.ts(2345)

Reply:
Oh, my bad I read this instead. 
https://github.com/zulip/zulip/blob/be59ef9c02ddbd99eed28828cc34393ca4eee45e/web/src/message_lists.ts#L34
thank you for pointing it out.